Add release notes for Metals v0.10.6 (#3104)
scalameta/metals: website/blog/2021-09-06-tungsten.md

---
author: tgodzik
title: Metals v0.10.6 - Tungsten
authorURL: https://twitter.com/<login>

> authorImageURL: https://github.com/<login>.png

Reviewer: `<login>`

Author: Fixed!

title: Metals v0.10.6 - Tungsten
authorURL: https://twitter.com/TomekGodzik
authorImageURL: https://github.com/tgodzik.png
---

We're happy to announce the release of Metals v0.10.6, which brings about new
Scala 3 versions and JDK 17 support in addition to some new interesting
features.

<table>
<tbody>
  <tr>
    <td>Commits since last release</td>
    <td align="center">173</td>
  </tr>
  <tr>
    <td>Merged PRs</td>
    <td align="center">74</td>
  </tr>
    <tr>
    <td>Contributors</td>
    <td align="center">16</td>
  </tr>
  <tr>
    <td>Closed issues</td>
    <td align="center">25</td>
  </tr>
  <tr>
    <td>New features</td>
    <td align="center">4</td>
  </tr>
</tbody>
</table>

For full details: https://github.com/scalameta/metals/milestone/41?closed=1

Metals is a language server for Scala that works with VS Code, Vim, Emacs,
Sublime Text and Eclipse. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from [Lunatech](https://lunatech.com) along with contributors from
the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

> - Adjust indentation on paste.

Reviewer 2: ```suggestion
- Automatically adjust indentation on paste.
```

Author: Fixed!

- New code actions.
- [Scala 3] support for Scala 3.0.2 and 3.1.0-RC1.
- Allow organize imports on unsaved code.
- JDK 17 support.

## Adjust Indentation on paste

Starting with Scala 3 developers can use optional braces, which we mentioned in
some of the previous release notes. One of the main problems with it is making

> sure that users are able to properly paste code into existing blocks. Previously
> we could have just pasted into a braced block and adjust indentation later, but
> now it really matters that indentation is changed properly in order to make to
> code compile correctly.

Reviewer 2: ```suggestion
sure that users are able to properly paste code into existing blocks. Previously
we could just paste into a braced blocks and adjusted the indentation later, but
now it really matters that indentation is changed properly in order to make to
code compile correctly.
```

Author: thanks! Changed.

To help with that [@Giggiux](https://github.com/Giggiux) adjusted the existing

> range formatting to account for indentation in the pasted blocks. Starting from

Reviewer 2: I believe this only works with editors that support range formatting correct? We may want to mention that.

Author: Good idea, added a mention at the end of the section.

### Replace () with {} in functions and vice versa

It's quite often that in Scala we might want to change to a simple lambda to a
multiline one or vice versa. This has previously been a manual chore, but now
whenever possible you can do it via a code action:

![replace](https://i.imgur.com/DTylGT9.gif)

## Miscellaneous

- Fix completions for definitions from other files in Ammonite scripts.
- Support for Apple M1 and JDK 17. [@dos65](https://github.com/dos65)
- [Scala 3] Show extension methods in hover.
- Support brackets in multiline strings.
  [@kpodsiad](https://github.com/kpodsiad)
- Automatically convert $ to $$ when changing to string interpolation.
- Filter out wrong reference results from unrelated build targets.
  [@dos65](https://github.com/dos65)
- Greatly improved parent code lenses performance and turned it on by default in
  VS Code. [@tanishiking](https://github.com/tanishiking)
- Properly show folding ranges in Scala 3 indented blocks.
  [@tpasternak](https://github.com/tpasternak)
- Fix comments erasing all later variables in .env file.
- Fix file watcher in large projects on macOS. [@pvid](https://github.com/pvid)

> 

Reviewer 2: What do you think about mentioning the memory fix that @dos65 did here? I know a lot of people have been mentioning issues with Scala 3 and memory issues, so it might be nice to mention.

Author: Added!
